static/app/components/searchQueryBuilder/useQueryBuilderState.tsx

```diff
@@ -78,8 +82,8 @@ export type QueryBuilderActions =
   | DeleteLastMultiSelectFilterValueAction;
 
 function removeQueryToken(query: string, token: TokenResult<Token>): string {
-  return (
-    query.substring(0, token.location.start.offset) +
+  return removeExcessWhitespaceFromParts(
```

Comment:
Some modifications in this file to trimming whitespace, because I was seeing that the stack would have duplicates of things like `is:unresolved browser.name:Firefox` and `is:unresolved  browser.name:Firefox` (multiple spaces in between). So I modified this to be more aggressive with enforcing a single space between tokens